tests/testing-framework/README.md

```diff
@@ -1,13 +1,48 @@
-# Runtime Tests
+# Testing Framework
 
+The testing framework is a general framework for running tests against a Spin compliant runtime. The testing framework includes the ability to start up dependent services (e.g., Redis or MySQL).
+
+## Services
+
+Services allow for tests to be run against external sources. The service definitions can be found in the 'services' directory. Each test directory contains a 'services' file that configures the tests services. Each line of the services file should contain the name of a services file that needs to run. For example, the following 'services' file will run the `tcp-echo.py` service:
+
+```txt
+tcp-echo
+```
+
+Each service is run under a file lock meaning that all other tests that require that service must wait until the current test using that service has finished.
+
+The following service types are supported:
+* Python services (a python script ending in the .py file extension)
```

Comment:
No changes needed here, but I just wanted to note that we could eventually support a third type of service using e.g. https://rust-script.org/, e.g. if there's useful Rust library providing a service we want to test against.

More generally, instead thinking of these as Python services, perhaps we could think of them as script services, allowing any of several extensions (.py, .rb, .rs, .sh, etc.) to be used, with the service author selecting the language according to the libraries available.

Reply:
Adding support for more types of scripting services is reasonable, but I think it's important that we limit the supported languages to ones that we can be generally sure are widely available across many different dev machines. Python is widely chosen because it's usually available on a dev's machine and if not, it is very easy to install. Other languages like bash might be common on some machines but a pain to install on others (e.g., on Windows outside of WSL contexts). 

So in general, I'm in favor of adding more support, but I'd like to make sure that we don't explode the choice of scripting language without an understanding of how it impacts how easy it is to run these tests.